### PR TITLE
Feat(swizzlemask): Allow one mask for an animation

### PR DESCRIPTION
**Pull request by RisingLeaf**
(requested by @Azure3141)
## Feature Details
If there are less swizzle-mask frames than animation frames, the last given swizzle-mask is used for the rest of the animation.

## Usage Examples
For the animation:
`ship-0.png`
`ship-1.png`
`ship-2.png`
`ship-3.png`

Then:
`ship@sw.png` would be used for the whole animation.

Or:
`ship-0@sw.png`
`ship-1@sw.png`

In which case `ship-1@sw.png` would be used for frame 2 and 3 too.

## Testing Done
Tested it on the vyur

## Performance Impact
N/A just quality of life

### CODE REVIEW

**Review by quyykk — COMMENTED** [review]
This is loading the same image multiple times over right? Is it possible to improve this without too many hacks?

**Comment by RisingLeaf** [comment]
> This is loading the same image multiple times over right? Is it possible to improve this without too many hacks?

As discussed shortly, probably not

**Comment by quyykk** [comment]
One problem with this feature is that it hides issues if you didn't mean to forget a swizzle mask in an animation sequence.

@Azure3141 Is it really that bad to copy a swizzle mask multiple times? :smile:

**Comment by quyykk** [comment]
@RisingLeaf Just adding the missing swizzle masks from the last one specified is a bit error prone, so IMO I think you should change this so that you have the option of 1) either specify every swizzle mask like how it is currently or 2) just specify one mask that would be used for the whole animation sequence.

**Comment by RisingLeaf** [comment]
> @RisingLeaf Just adding the missing swizzle masks from the last one specified is a bit error prone, so IMO I think you should change this so that you have the option of 1) either specify every swizzle mask like how it is currently or 2) just specify one mask that would be used for the whole animation sequence.

and where should that option be?
sprites are loaded independent from data

**Comment by quyykk** [comment]
No, just use the mask at animation index 0, like `img-00@sw.png`

**Comment by RisingLeaf** [comment]
Sorry I still fail to understand how you imagine this to work

**Comment by quyykk** [comment]
If there is only one mask, and that mask is for the first frame, make it apply to every frame.

**Comment by RisingLeaf** [comment]
It should do that now

File: source/ImageSet.cpp [hunk 1]
```diff
@@ -283,6 +283,17 @@ void ImageSet::Load() noexcept(false)
 		}
 	}
 
+	auto FillSwizzleMasks = [&](vector<string> &toFill, unsigned int intendedSize) {
+		if(toFill.size() == 1 && intendedSize > 1)
```
Comment (quyykk) [inline source/ImageSet.cpp:287]: Do check if this is the first frame as well.

**Review by quyykk — APPROVED** [review]

**Pull request merged**